### DISCUSSION TITLE
DataTable - Conditional Expander

### DISCUSSION THREAD

**Discussion by mkalinski93** [primefaces/community#2977] [PrimeVue]
Hey,

I just wanted to ask if there´s an inbuilt method that I can show/hide the row expander conditionally.

If there´s no provided method, would something like that be possible?

<Column>
    <template #expansion={data: currentRow, enable: boolean }/>
</Column>

**Comment by j0rgedev** [comment]
Hello @mkalinski93. You can actually expand some rows by default. If you look at this [example](https://stackblitz.com/run?file=src%2FApp.vue). There is a ref that tracks the expanded rows: `const expandedRows = ref({});`, which is an object that takes the id as the key, and a boolean as its value. So, if you want to expand some rows conditionally, you could play with that.
Based on the example, imagine you want to expand the rows where the price is greater than 100

```js
... more code
const products = ref();
const expandedRows = ref({});

onMounted(() => {
  ProductService.getProductsWithOrdersSmall().then((data) => {
    products.value = data;
    
    expandedRows.value = data.reduce((acc, product) => {
      if (product.price > 100) {
        acc[product.id] = true;
      }
      return acc;
    }, {});
  });
});

... more code
```
